File: tests/fixtures/lang_samples/typescript/assertions.ts
// Fixture for assertion-block detection (test-quality smells).

function testManyExpects(): void {
  const x = 1;
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
  expect(x).toBe(1);
}

function testFewExpects(): void {
  expect(1).toBe(1);
  const y = 2;
  expect(y).toBe(2);
}

File: tests/fixtures/lang_samples/python/assertions.py
"""Fixture for assertion-block detection (test-quality smells)."""


def test_many_bare_asserts():
    x = 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1
    assert x == 1


class _Case:
    def test_unittest_calls(self):
        self.assertEqual(1, 1)
        self.assertEqual(2, 2)
        self.assertTrue(True)

    def test_few_asserts(self):
        assert 1 == 1
        x = 2
        assert x == 2


def test_split_runs():
    assert 1 == 1
    assert 2 == 2
    do_work()
    assert 3 == 3
    assert 4 == 4


def do_work():
    return None
